src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+<Sandpack>
+
+```js message.js active
+"use client";
+
+import { use } from "react";
+
+export function Message({ messagePromise }) {
+  const messageContent = use(messagePromise);
+  return <p>Here is the message: {messageContent}</p>;
+}
+```
+
+```js App.js
+import { Suspense, useState } from "react";
+import { Message } from "./message.js";
+
+function fetchMessage() {
+  return new Promise((resolve) => setTimeout(resolve, 3000, "⚛️"));
+}
+
+export default function App() {
+  const [messagePromise, setMessagePromise] = useState(null);
+  const [show, setShow] = useState(false);
+  function download() {
+    setMessagePromise(fetchMessage());
+    setShow(true);
+  }
+
+  if (show) {
+    return (
+      <Suspense fallback={<p>⌛Downloading message...</p>}>
+        <Message messagePromise={messagePromise} />
+      </Suspense>
+    );
+  } else {
+    return <button onClick={download}>Download message</button>;
+  }
+}
+```
+
+```js index.js hidden
+// TODO: update to import from stable
+// react instead of canary once the `use`
+// Hook is in a stable release of React
+import React, { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
+import "./styles.css";
+
+// TODO: update this example to use
+// the Codesandbox Server Component
+// demo enviornemnt once it is created
+import App from "./App";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+<DeepDive>
+
+#### Server components: `await` vs passing a promise? {/*await-vs-passing-a-promise*/}
+
+Instead of passing a promise from a server component to a client component you could `await` the promise in the server component and pass the data to the client component as a prop:
+
+```js
+import { fetchMessage } from "./lib.js";
+import { Message } from "./message.js";
+
+export default function App() {
+  const messageContent = await fetchMessage();
+  return (
+    <Suspense fallback={<p>waiting for message...</p>}>
+      <Message messageContent={messageContent} />
+    </Suspense>
+  );
+}
+```
+
+But using `await` in a [Server Component](/reference/react/components#server-components) will block the rendering of the [Server Component](/reference/react/components#server-components) until the `await` statement is finished.
+
+Passing a promise from a server component to a client component prevents the promise from blocking the rendering of the server component. This enables a suspense fallback to be displayed to the user. When the promise resolves, the value of the promise is read by `use`. This value is used to render the client component which replaces the suspense fallback.
+
```

Comment:
What about if it was split so that Suspense was in a parent component?

Reply:
Added a note that you need to wrap the client component that is passed the promise in Suspense for this to work. Is that enough or would you think a different approach would be better?